ocaml/database/db_cache_types.ml

```diff
@@ -16,6 +16,20 @@ open Db_exn
 
 module Time = struct type t = Generation.t end
 
+module HashedString = struct
+  type t = string
+
+  let equal = String.equal
+
+  let hash = Hashtbl.hash
```

Comment:
Not sure it's worth, but you can have an extra perf improvement taking inspiration from https://github.com/ocaml/ocaml/pull/8878

Reply:
We are on OCaml 4.08 – I'd rather wait for `String.hash` or similar in the standard library than trying to do something clever here.